File: original_scripts/c/matrix_multiplication/qwen3_1.7b.c
#include <stdio.h>

void matrix_multiply(int* a, int* b, int* c, int rows_a, int cols_a, int rows_b) {
    for (int i = 0; i < rows_a; i++) {
        for (int j = 0; j < cols_a; j++) {
            int sum = 0;
            for (int k = 0; k < rows_b; k++) {
                sum += a[i * rows_b + k] * b[j * rows_b + k];
            }
            c[i * cols_a + j] = sum;
        }
    }
}

File: original_scripts/rust/matrix_multiplication/qwen3_1.7b.rs
fn matrix_mult(a: &[Vec<i32>, b: &[Vec<i32>]) -> Vec<Vec<i32> > {
    let rows_a = a.len();
    let cols_a = a[0].len();
    let rows_b = b.len();
    let cols_b = b[0].len();

    let mut result = vec![vec![]; rows_a];

    for i in 0..rows_a {
        for j in 0..cols_b {
            let mut val = 0;
            for k in 0..cols_a {
                val += a[i][k] * b[k][j];
            }
            result[i][j] = val;
        }
    }

    result
}